examples/features/observability/go.mod

```diff
@@ -0,0 +1,47 @@
+module google.golang.org/grpc/examples/features/observability
+
+go 1.17
+
+require (
+	google.golang.org/grpc v1.54.0
+	google.golang.org/grpc/examples v0.0.0-20230323213306-0fdfd40215dc
+	google.golang.org/grpc/gcp/observability v0.0.0-00010101000000-000000000000
+)
+
+require (
+	cloud.google.com/go v0.109.0 // indirect
+	cloud.google.com/go/compute v1.18.0 // indirect
+	cloud.google.com/go/compute/metadata v0.2.3 // indirect
+	cloud.google.com/go/logging v1.6.1 // indirect
+	cloud.google.com/go/longrunning v0.4.0 // indirect
+	cloud.google.com/go/monitoring v1.12.0 // indirect
+	cloud.google.com/go/trace v1.8.0 // indirect
+	contrib.go.opencensus.io/exporter/stackdriver v0.13.12 // indirect
+	github.com/aws/aws-sdk-go v1.44.162 // indirect
+	github.com/census-instrumentation/opencensus-proto v0.4.1 // indirect
+	github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da // indirect
+	github.com/golang/protobuf v1.5.3 // indirect
+	github.com/google/go-cmp v0.5.9 // indirect
+	github.com/google/uuid v1.3.0 // indirect
+	github.com/googleapis/enterprise-certificate-proxy v0.2.3 // indirect
+	github.com/googleapis/gax-go/v2 v2.7.0 // indirect
+	github.com/jmespath/go-jmespath v0.4.0 // indirect
+	github.com/prometheus/prometheus v2.5.0+incompatible // indirect
+	go.opencensus.io v0.24.0 // indirect
+	golang.org/x/net v0.8.0 // indirect
+	golang.org/x/oauth2 v0.5.0 // indirect
+	golang.org/x/sync v0.1.0 // indirect
+	golang.org/x/sys v0.6.0 // indirect
+	golang.org/x/text v0.8.0 // indirect
+	google.golang.org/api v0.109.0 // indirect
+	google.golang.org/appengine v1.6.7 // indirect
+	google.golang.org/genproto v0.0.0-20230125152338-dcaf20b6aeaa // indirect
+	google.golang.org/grpc/stats/opencensus v0.0.0-20230317183452-b638faff2204 // indirect
+	google.golang.org/protobuf v1.29.1 // indirect
+)
+
+replace google.golang.org/grpc => ../../..
+
+replace google.golang.org/grpc/gcp/observability => ../../../gcp/observability
+
+replace google.golang.org/grpc/stats/opencensus => ../../../stats/opencensus
```

Comment:
I think we should remove these so the go.mod is as clean as possible (i.e. closest to what users should be using).